DOCUMENTATION.md

```diff
@@ -104,751 +29,740 @@ import uk.gov.service.notify.SendSmsResponse;
+```
+
+If you do not have have an `emailReplyToId`, you must pass in an empty string or `null`.
+
+### Response
+
+If the request to the client is successful, you will receive the following `SendEmailResponse`:
 
 ```java
-Map<String, String> personalisation = new HashMap<>();
-personalisation.put("name", "Jo");
-personalisation.put("reference_number", "13566");
+UUID notificationId;
+Optional<String> reference;
+UUID templateId;
+int templateVersion;
+String templateUri;
+String body;
+String subject;
+Optional<String> fromEmail;
 ```
 
-#### `reference`
-An optional identifier you generate. The reference can be used as a unique reference for the notification. Because Notify does not require this reference to be unique you could also use this reference to identify a batch or group of notifications.
+### Error codes
+
+If the request is not successful, the client will raise an `HTTPError`:
 
-You can omit this argument if you do not require a reference for the notification.
+|`httpResult`|`message`|How to fix|
+|:---|:---|:---|
+|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Can't send to this recipient using a team-only API key"`<br>`]}`|Use the correct type of API key. Refer to [API keys](/#api-keys) for more information|
+|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Can't send to this recipient when service is in trial mode - see https://www.notifications.service.gov.uk/trial-mode"`<br>`}]`|Refer to [trial mode](https://www.notifications.service.gov.uk/features/using-notify#trial-mode) for more information|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Error: Your system clock must be accurate to within 30 seconds"`<br>`}]`|Check your system clock|
+|`403`|`[{`<br>`"error": "AuthError",`<br>`"message": "Invalid token: signature, api token not found"`<br>`}]`|Use the correct API key. Refer to [API keys](/#api-keys) for more information|
+|`429`|`[{`<br>`"error": "RateLimitError",`<br>`"message": "Exceeded rate limit for key type TEAM/TEST/LIVE of 3000 requests per 60 seconds"`<br>`}]`|Refer to [API rate limits](/#api-rate-limits) for more information|
+|`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (LIMIT NUMBER) for today"`<br>`}]`|Refer to [service limits](/#service-limits) for the limit number|
+|`500`|`[{`<br>`"error": "Exception",`<br>`"message": "Internal server error"`<br>`}]`|Notify was unable to process the request, resend your notification|
 
-#### `emailReplyToId`
-Optional. Specifies the identifier of the email reply-to address to set for the notification. The identifiers are found in your service Settings, when you 'Manage' your 'Email reply to addresses'.
-If you omit this argument your default email reply-to address will be set for the notification.
+## Send a letter
 
-</details>
+When your service first signs up to GOV.UK Notify, you’ll start in trial mode. You can only send letters in live mode.
```

Comment:
"You can only send letters in live mode." ==> "You can only send letters once you are live. But you can test letters with a test key"

Reply:
Does this apply to Python as well?